homeassistant/components/google_assistant/helpers.py

```diff
@@ -120,14 +120,17 @@ def async_disable_report_state(self):
             self._unsub_report_state()
             self._unsub_report_state = None
 
-    async def async_sync_entities(self):
+    async def async_sync_entities(self, agent_user_id: Optional[str] = None):
         """Sync all entities to Google."""
         # Remove any pending sync
         if self._google_sync_unsub:
             self._google_sync_unsub()
             self._google_sync_unsub = None
 
-        return await self._async_request_sync_devices()
+        if agent_user_id is None:
```

Comment:
Let's remove support for defaulting this to `None` and just have cloud pass in `config.agent_user_id`.

Reply:
I'm allow None here as prepatory for this: https://github.com/elupus/home-assistant/compare/google_assistant_storage...elupus:google_assistant_storage_2?expand=1#diff-3b8968e0e00d5858a42a853371fcfeb7R135

That is allowing a sync request for a specific agent, rather than for all previously synced agents.

Reply:
So i don't mind changing it. But i might need to change it back. The alternative option is to introduce a async_sync_entities_all() without a parameter for syncing all in above change.

Reply:
@balloob could i get a comment on above. The point is when we take next step we need a way to update a specific agent_user_id (to retain the old service) and support syncing all connected users.